src/Controls/GlobeControls.js

```diff
@@ -131,8 +130,28 @@ export const CONTROL_EVENTS = {
     CAMERA_TARGET_CHANGED: 'camera-target-changed',
 };
 
+const quaterPano = new THREE.Quaternion();
+const quaterAxis = new THREE.Quaternion();
+const axisX = new THREE.Vector3(1, 0, 0);
+let minDistanceZ = Infinity;
+const lastNormalizedIntersection = new THREE.Vector3();
+const normalizedIntersection = new THREE.Vector3();
+const raycaster = new THREE.Raycaster();
+const targetPosition = new THREE.Vector3();
+const pickedPosition = new THREE.Vector3();
+
 let previous;
 
+function getRangeFromScale(scale, pitch, fov, height) {
+    // Screen pitch, in millimeters
+    pitch = (pitch || 0.28) / 1000;
+    fov = THREE.Math.degToRad(fov);
+    // Invert one unit projection (see getDollyScale)
+    const range = pitch * height / (scale * 2 * Math.tan(fov * 0.5));
+
+    return range;
+}
+
```

Comment:
Maybe you could replace this by `View#getScaleFromDistance`, or at least use it.

Reply:
It seems you can also replace things like `getRange` by `View#getDistanceFromCamera`. There seems to be quite a lot of method that are giving the same result while doing different things, so maybe take a look into that if you have time (between `View`, `CameraUtils` and `GlobeControls` for instance).

Reply:
Yes, I'll create a new PR to re-factorize and I only use the class in this PR.